stack.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python
+
+
+class Element(object):
+    ''' Create data element with default value and previous pointer. '''
+    def __init__(self, value=None, previous=None):
+        ''' Value and previous pointer default to none. '''
+        self.val = value
+        self.previous = previous
+
+
+class Stack(object):
+    ''' Create an empty stack. '''
+    def __init__(self):
+        self.top = None
+
+    def push(self, element):
+        ''' Add data element to the top of stack. '''
+        self.top, self.top.previous = element, self.top
+
+    def pop(self):
+        ''' Remove top element from stack. Reassign and reassign top data
+            element. '''
+        if not self.top:
```

Comment:
This is a place where "it is better to ask forgiveness than permission" could come in handy.  Don't bother checking if self.top exists, just try to get the value and deal with the `AttributeError` if it doesn't.